### PR TITLE
Bug fix: allow authentication addons to create users again.

### PR DESCRIPTION
**Pull request by very-ape**
In testing I discovered that the change I made broke user creation in the LDAP addon. This should fix it.

### CODE REVIEW

File: src/Model/User.php [hunk 1]
```diff
@@ -515,7 +515,28 @@ public static function authenticate($user_info, $password, $third_party = false)
...
+			if (is_numeric($user_info) || is_numeric($user_info['nickname'] ?? '')) {
+				throw $e;
+			}
+
+			$username = (is_string($user_info) ? $user_info : $user_info['nickname'] ?? '');
+
+			if (!$username) {
+				throw $e;
+			}
+			return self::getIdFromAuthenticateHooks($user_info, $password);
```
Comment (MrPetovan) [inline src/Model/User.php]: Code standards.
```suggestion
			$username = (is_string($user_info) ? $user_info : $user_info['nickname'] ?? '');
			if (!$username) {
				throw $e;
			}

			return self::getIdFromAuthenticateHooks($user_info, $password);
```

File: src/Model/User.php [hunk 2]
```diff
@@ -545,23 +566,39 @@ public static function getIdFromPasswordAuthentication($user_info, $password, $t
...
+	 * Try to obtain a user ID via "authenticate" hook addons
+	 *
+	 * Returns the user id associated with a successful password authentication
+	 *
+	 * @param string $username
+	 * @param string $password
+	 * @return int User Id if authentication is successful
+	 * @throws HTTPException\ForbiddenException
+	 */
+	public static function getIdFromAuthenticateHooks($username, $password) {
```
Comment (MrPetovan) [inline src/Model/User.php]: Code standards.
```suggestion
	public static function getIdFromAuthenticateHooks($username, $password)
	 {
```

**Review by MrPetovan — CHANGES_REQUESTED** [review]
Thank you for the comments!

**Comment by very-ape** [comment]
No problem! Thanks for the quick reviews. :)

**Comment by very-ape** [comment]
Crap, accidentally committed changes just now that weren't intended for this PR. One moment while I fix that.

**Comment by very-ape** [comment]
Fixed!

File: src/Model/User.php [hunk 3]
```diff
@@ -515,7 +515,27 @@ public static function authenticate($user_info, $password, $third_party = false)
...
+			$user = self::getAuthenticationInfo($user_info);
+		} catch (Exception $e) {
+			$username = (is_string($user_info) ? $user_info : $user_info['nickname'] ?? '');
+
+			// Addons can create users, and since this 'catch' branch should only
+			// execute if getAuthenticationInfo can't find an existing user, that's
+			// exactly what will happen here. Creating a numeric username would create
+			// abiguity with user IDs, possibly opening up an attack vector.
+			// So let's be very careful about that.
+			if (empty($username) || is_numeric($user_info) || is_numeric($user_info['nickname'] ?? '')) {
```
Comment (MrPetovan) [inline src/Model/User.php]: `is_numeric($user_info)` already is checked on line 526, and `$user_info['nickname'] ?? ''` already is assigned to `$username`, so this condition can be simplified.
```suggestion
			if (empty($username) || is_numeric($username)) {
```

**Review by MrPetovan — CHANGES_REQUESTED** [review]

File: src/Model/User.php (hunk 3 above)
Comment (very-ape) [reply src/Model/User.php]: Yup, good eye.

**Review by MrPetovan — APPROVED** [review]

**Review by MrPetovan — APPROVED** [review]

**Comment by MrPetovan** [comment]
Follow-up to https://github.com/friendica/friendica/pull/10275

**Pull request merged**